wandb/integration/diffusers/resolvers/sdxl.py

```diff
@@ -0,0 +1,268 @@
+
+            # Return the WandB loggable dict
+            loggable_dict = self.prepare_loggable_dict(
+                pipeline, pipeline_configs, response, kwargs
+            )
+            return loggable_dict
+        except Exception as e:
+            logger.warning(e)
+        return None
+
+    def create_wandb_table(self, pipeline_configs: Dict[str, Any]) -> None:
+        """Create the `wandb.Table` with the specified schema.
+
+        Arguments:
+            pipeline_configs: (Dict[str, Any]) The configs corresponding to the
+                architecture of the pipeline.
+        """
+        columns = []
+        if self.pipeline_name == "StableDiffusionXLPipeline":
+            columns += TEXT_TO_IMAGE_COLUMNS
+            self.task = "text_to_image"
+            self.workflow_stage = "Base"
+        elif self.pipeline_name == "StableDiffusionXLImg2ImgPipeline":
+            if (
+                pipeline_configs["_name_or_path"]
+                == "stabilityai/stable-diffusion-xl-refiner-1.0"
+            ):
+                columns += TEXT_TO_IMAGE_COLUMNS
+                self.task = "text_to_image"
+                self.workflow_stage = "Refiner"
+        self.wandb_table = wandb.Table(columns=columns)
+
+    def prepare_loggable_dict_for_text_to_image(
+        self,
+        pipeline: Any,
+        workflow_stage: str,
+        response: Response,
+        kwargs: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        """Prepare the loggable dictionary for text-to-image workflow.
+
+        Arguments:
+            pipeline: (Any) The Diffusion Pipeline.
+            workflow_stage: (str) The name of the workflow stage.
+            response: (wandb.sdk.integration_utils.auto_logging.Response) The response from
+                the request.
+            kwargs: (Dict[str, Any]) Dictionary of keyword arguments.
+
+        """
+        prompt_logging = (
+            kwargs["prompt"]
+            if isinstance(kwargs["prompt"], list)
+            else [kwargs["prompt"]]
+        )
+        prompt2_logging = (
+            kwargs["prompt_2"]
+            if isinstance(kwargs["prompt_2"], list)
+            else [kwargs["prompt_2"]]
+        )
+        negative_prompt_logging = (
+            kwargs["negative_prompt"]
+            if isinstance(kwargs["negative_prompt"], list)
+            else [kwargs["negative_prompt"]]
+        )
+        negative_prompt2_logging = (
+            kwargs["negative_prompt_2"]
+            if isinstance(kwargs["negative_prompt_2"], list)
+            else [kwargs["negative_prompt_2"]]
+        )
+        images = (
+            decode_sdxl_t2i_latents(pipeline, response.images)
+            if kwargs["output_type"] == "latent"
+            else response.images
+        )
+        images = chunkify(images, len(prompt_logging))
+        for idx in range(len(prompt_logging)):
+            for image in images[idx]:
+                wandb.log(
```

Comment:
are you guaranteed to have a run when calling this?

Reply:
Yes, a run is guaranteed to be initialized before a resolver is called.